yrs/src/block_store.rs

```diff
@@ -425,29 +425,25 @@ impl BlockStore {
         clients.try_get(pivot)
     }
 
-    pub(crate) fn get_item_clean_start(&mut self, id: &ID) -> Option<BlockPtr> {
-        let blocks = self.clients.get_mut(&id.client)?;
-        let mut index = blocks.find_pivot(id.clock)?;
-        let mut ptr = blocks.get(index);
-        if let Some(new) = ptr.splice(id.clock - ptr.id().clock, OffsetKind::Utf16) {
-            blocks.insert(index + 1, new);
-            index += 1;
-        }
-        Some(blocks.get(index))
+    pub(crate) fn get_item_clean_start(&self, id: &ID) -> Option<BlockSlice> {
+        let blocks = self.clients.get(&id.client)?;
+        let index = blocks.find_pivot(id.clock)?;
+        let ptr = blocks.get(index);
+        let offset = id.clock - ptr.id().clock;
+        Some(BlockSlice::new(ptr, offset, ptr.len() - 1))
     }
 
-    pub(crate) fn get_item_clean_end(&mut self, id: &ID) -> Option<BlockPtr> {
-        let blocks = self.clients.get_mut(&id.client)?;
+    pub(crate) fn get_item_clean_end(&self, id: &ID) -> Option<BlockSlice> {
```

Comment:
`get_item_clean_start` and `get_item_clean_end` now are returning `BlockSlice` which means, that they don't split blocks right away. Store has a dedicated method that can split block along the slice boundaries when necessary.